src/IECoreScene/ShaderNetworkAlgo.cpp

```diff
@@ -684,6 +713,213 @@ IECore::DataPtr loadSpline(
 
 } // namespace
 
+void ShaderNetworkAlgo::convertToOSLConventions( ShaderNetwork *network, int oslVersion )
+{
+	std::map< IECore::InternedString, std::pair< IECore::InternedString, size_t > > currentSplineArrayAdapters;
+
+	for( const auto &s : network->shaders() )
+	{
+		if( !boost::starts_with( s.second->getType(), "osl:" ) )
+		{
+			continue;
+		}
+
+		const auto &origParameters = s.second->parameters();
+		bool hasSplines = false;
+		for( const auto &i : origParameters )
+		{
+			if(
+				runTimeCast<const SplinefColor3fData>( i.second.get() ) ||
+				runTimeCast<const SplineffData>( i.second.get() )
+			)
+			{
+				hasSplines = true;
+				break;
+			}
+		}
+
+		if( !hasSplines )
+		{
+			continue;
+		}
+
+		CompoundDataPtr newParametersData = new CompoundData();
+		CompoundDataMap &newParameters = newParametersData->writable();
+
+		for( const auto &i : origParameters )
```

Comment:
I'm a fan of structured bindings for this sort of thing, so you can have meaningful names instead of `i.first` and `i.second` :

```
for( const auto &[name, value] : origParameters )
```